internal/raft/message/convert.go

```diff
@@ -0,0 +1,971 @@
+	case *command.RangeExpr:
+		msgExpr.Expr = ConvertCommandToMessageRangeExpr(cmd)
+	}
+	return msgExpr
+}
+
+// ConvertCommandToMessageListScan converts a command.Scan to a message.List_Scan
+func ConvertCommandToMessageListScan(cmd *command.Scan) *List_Scan {
+	msgListScan := &List_Scan{&Command_Scan{}}
+	msgListScan.Scan.Table = ConvertCommandToMessageTable(cmd.Table)
+	return msgListScan
+}
+
+// ConvertCommandToMessageListSelect converts a command.Select to a message.List_Select
+func ConvertCommandToMessageListSelect(cmd *command.Select) *List_Select {
+	msgListSelect := &List_Select{&Command_Select{}}
+	msgListSelect.Select.Filter = ConvertCommandToMessageExpr(cmd.Filter)
+	msgListSelect.Select.Input = ConvertCommandToMessageList(cmd.Input)
+	return msgListSelect
+}
+
+// ConvertCommandToMessageListProject converts a command.Project to a message.List_Project
+func ConvertCommandToMessageListProject(cmd *command.Project) *List_Project {
+	msgListProject := &List_Project{&Command_Project{}}
+	msgListProject.Project.Cols = ConvertCommandToMessageCols(cmd.Cols)
+	msgListProject.Project.Input = ConvertCommandToMessageList(cmd.Input)
+	return msgListProject
+}
+
+// ConvertCommandToMessageListJoin converts a command.Join to a message.List_Join
+func ConvertCommandToMessageListJoin(cmd *command.Join) *List_Join {
+	msgListJoin := &List_Join{&Command_Join{}}
+	msgListJoin.Join.Natural = cmd.Natural
+	msgListJoin.Join.Type = ConvertCommandToMessageJoinType(cmd.Type)
+	msgListJoin.Join.Filter = ConvertCommandToMessageExpr(cmd.Filter)
+	msgListJoin.Join.Left = ConvertCommandToMessageList(cmd.Left)
+	msgListJoin.Join.Right = ConvertCommandToMessageList(cmd.Right)
+	return msgListJoin
+}
+
+// ConvertCommandToMessageListLimit converts a command.Limit to a message.List_Limit
+func ConvertCommandToMessageListLimit(cmd *command.Limit) *List_Limit {
+	msgListLimit := &List_Limit{&Command_Limit{}}
+	msgListLimit.Limit.Limit = ConvertCommandToMessageExpr(cmd.Limit)
+	msgListLimit.Limit.Input = ConvertCommandToMessageList(cmd.Input)
+	return msgListLimit
+}
+
+// ConvertCommandToMessageListOffset converts a command.Offset to a message.List_Offset
+func ConvertCommandToMessageListOffset(cmd *command.Offset) *List_Offset {
+	msgListOffset := &List_Offset{&Command_Offset{}}
+	msgListOffset.Offset.Offset = ConvertCommandToMessageExpr(cmd.Offset)
+	msgListOffset.Offset.Input = ConvertCommandToMessageList(cmd.Input)
+	return msgListOffset
+}
+
+// ConvertCommandToMessageListDistinct converts a command.Distinct to a message.List_Distinct
+func ConvertCommandToMessageListDistinct(cmd *command.Distinct) *List_Distinct {
+	msgListDistinct := &List_Distinct{&Command_Distinct{}}
+	msgListDistinct.Distinct.Input = ConvertCommandToMessageList(cmd.Input)
+	return msgListDistinct
+}
+
+// ConvertCommandToMessageRepeatedExprSlice converts a [][]command.Expr to a [][]message.Expr
+func ConvertCommandToMessageRepeatedExprSlice(cmd [][]command.Expr) []*RepeatedExpr {
```

Comment:
Why does it not take a `Command` if the name says "convert command to message..."?

Reply:
It does take a command of type `command.Expr`. Also its called by a higher level conversion of a `command.Command1

Reply:
It takes a `[][]command.Expr`, which is not a `command.Command`